supabase/migrations/33_automate_trial_start.sql

```diff
@@ -0,0 +1,88 @@
+begin;
+
+create or replace view internal.new_free_trial_tenants as
+with hrs_by_day as (
+  select
+    tenants.tenant as tenant,
+    ts,
+    sum(catalog_stats_daily.usage_seconds / (60.0 * 60)) as daily_usage_hours
+  from catalog_stats_daily
+  join tenants on tenants.tenant = split_part(catalog_stats_daily.catalog_name,'/',1)||'/'
```

Comment:
always always use the `^@` operator for prefix comparisons. Compare to how we're doing it elsewhere for authorization checks, etc.

The reason why is because this usage of split_part forces it to walk the table actually doing string splits for every row. The optimizer almost certainly can't "see into" what this is doing. Whereas postgres is smart enough to push down `^@` into an evaluation against the ordered table index.

Reply:
So yeah I must be doing something wrong then. I originally used `^@`, but switched to this much less legible `split_part` filtering because it's _dramatically_ more performant. Just switching the 3 join conditions in the CTEs to `join tenants on catalog_stats.catalog_name ^@ tenants.tenant` instead of `split_parts` causes the query to take 117s, instead of 500ms. Here's just one of the CTEs split out:

```sql
 select
    tenants.tenant as tenant,
    ts,
    sum(catalog_stats_daily.usage_seconds / (60.0 * 60)) as daily_usage_hours
  from catalog_stats_daily
  join tenants on catalog_stats_daily.catalog_name ^@ tenants.tenant
  -- join tenants on tenants.tenant = split_part(catalog_stats_daily.catalog_name,'/',1)||'/'
  where tenants.trial_start is null
  group by tenants.tenant, ts
  having sum(catalog_stats_daily.usage_seconds / (60.0 * 60)) > (2 * 24)
```
* With `split_part` it runs in 312ms. `EXPLAIN` output: https://explain.dalibo.com/plan/4051a76d37ag4b2h
* With `^@`, it runs in **1m35s**. `EXPLAIN` output: https://explain.dalibo.com/plan/he33e57c8a07702f

Reply:
Thx for your help (on slack, not being sassy here lolol). Turns out postgres is finicky and won't (or can't) use the regular btree index that we had on `catalog_stats.catalog_name`. The following index fixed it, and is included in this migration (though I already applied it to the DB):

```sql
create index catalog_stats_catalog_index_spgist on catalog_stats using spgist ((catalog_name::text));
```